packages/synapse-react-client/src/utils/functions/ArrayUtils.test.ts

```diff
@@ -26,4 +26,15 @@ describe('ArrayUtils', () => {
   ])('removeTrailingUndefinedElements - %s', (caseName, arg, expected) => {
     expect(removeTrailingUndefinedElements(arg)).toEqual(expected)
   })
+
+  test.each([
+    [[], [], false],
+    [[1, 2, 3], [1, 2, 3], false],
+    [[1, 2, 3], [1, 2, 4], true],
+    [[1, 2, 3], [1, 2], true],
```

Comment:
I would also verify that order does not matter.  I think this will pass given your current implementation (using _xorWith):
[[1, 2], [2, 1], false],